File: backend/app/schemas/resume.py
from pydantic import BaseModel, Field

from app.schemas.report import LearningRoadmapItem, ResourceRecommendation


class ResumeAnalysisTextRequest(BaseModel):
    resume_text: str
    target_role: str


class ResumeAnalysisResponse(BaseModel):
    target_role: str
    matching_score: float
    ats_readiness_score: float = 0
    role_fit_summary: str = ""
    detected_skills: list[str] = Field(default_factory=list)
    missing_skills: list[str] = Field(default_factory=list)
    priority_gaps: list[str] = Field(default_factory=list)
    evidence_highlights: list[str] = Field(default_factory=list)
    extracted_text_preview: str = ""
    learning_roadmap: list[LearningRoadmapItem] = Field(default_factory=list)
    recommendations: list[ResourceRecommendation] = Field(default_factory=list)
    personalized_coaching: list[str] = Field(default_factory=list)
    report_id: str | None = None
    artifact_relative_path: str | None = None

File: frontend/lib/resume.ts
import { apiRequest } from "@/lib/api";

export type ResourceRecommendation = {
  topic: string;
  title: string;
  platform: string;
  price: string;
  url: string;
};

export type LearningRoadmapItem = {
  skill: string;
  next_step: string;
  resource_title: string | null;
  resource_url: string | null;
};

export type ResumeAnalysisResponse = {
  target_role: string;
  matching_score: number;
  ats_readiness_score: number;
  role_fit_summary: string;
  detected_skills: string[];
  missing_skills: string[];
  priority_gaps: string[];
  evidence_highlights: string[];
  extracted_text_preview: string;
  learning_roadmap: LearningRoadmapItem[];
  recommendations: ResourceRecommendation[];
  personalized_coaching: string[];
  report_id: string | null;
  artifact_relative_path: string | null;
};

export async function analyzeResumeText(
  token: string,
  payload: { resume_text: string; target_role: string }
) {
  return apiRequest<ResumeAnalysisResponse>("/resume-analysis/text", {
    method: "POST",
    token,
    body: JSON.stringify(payload)
  });
}

export async function analyzeResumeUpload(
  token: string,
  payload: { file: File; targetRole: string }
) {
  const formData = new FormData();
  formData.append("resume", payload.file);
  formData.append("target_role", payload.targetRole);

  return apiRequest<ResumeAnalysisResponse>("/resume-analysis/upload", {
    method: "POST",
    token,
    body: formData
  });
}
